Update pip-tools configuration for OEP-18

michaelyoungstrom opened the pull request.

jmbowman commented on Makefile:
```diff
@@ -1,22 +1,26 @@
-	pip-sync dev-requirements.txt requirements.txt
+dev-install:
+	pip install -r requirements/base.txt
+	pip install -r requirements/development.txt
```
requirements/development.in should include requirements/base.txt so the package versions stay consistent; then you'll only need to install the second file here, since the first will be a subset of it.

jmbowman commented on post-pip-compile.sh:
```diff
@@ -0,0 +1,34 @@
+    echo ""
+    echo "Updates the instructions for re-generating each requirements file, and removes"
+    echo "\"-e\" prefixes which were added to GitHub URLs only so that pip-compile could"
+    echo "process them correctly."
```
The help text should probably be updated to reflect what this repo's copy of the script actually does.

michaelyoungstrom commented:
@jmbowman I think I've addressed all of your feedback now.

I'm curious to see what you think of the gitgraft requirements file. I have the `.in`file inside the `requirements` dir, but the `make upgrade` command creates the `.txt` file in the location it's always been. Would you prefer me to put a stub file there and have it be created in the same `requirements` dir as the others? I'd be happy to change that if so.

Also, I pinned github3... there's been some major changes in the new release.

jmbowman commented on dev-requirements.txt:
```diff
@@ -1,35 +1 @@
-singledispatch==3.4.0.3   # via astroid, pylint
-six==1.10.0               # via astroid, edx-lint, mock, pip-tools, pylint, responses, singledispatch
-wrapt==1.10.10            # via astroid
+-r requirements/development.txt
```
Maybe add a brief comment explaining the plan to remove this and the criteria for doing so?  Ditto for requirements.txt.
